build_package.py

```diff
@@ -795,14 +826,41 @@ def build_base_files(base_configs="config"):
     os.chdir("py")
 
     shutil.copytree(make_path(agent_source_root, "scalyr_agent"), "scalyr_agent")
+
+    # patching "agent_main.py" entrypoint to enable coverage.
```

Comment:
Hmm.. patching the agent_main.py file seems more risky here.  Why not just patch the Dockerfile to replace the `CMD` line with what you had it in the coverage-specific Dockerfile files?    It was something like `CMD ["coverage", "run", "/usr/share/scalyr-agent-2/py/scalyr_agent/agent_main.py", "--no-fork", "--no-change-user", "start"]`.

After all, you are already patching the Dockerfile.  It seems better to just patch one thing.

Reply:
It will be great if it works without this when `docker-json` and `k8s` will start working again. I had some problems when I tried to get coverage results after agent termination and I was able to solve this only by enabling coverage from code.

Let's see when those jobs will work, maybe there is no need for this.
I removed `agent_main.py` patching and returned Dockerfile CMD patching. .